### PR TITLE
Toggle Button & Toggle Button Group

### LINKED ISSUE #597: Toggle Button Group and Button height difference
Toggle Button and Button are having different height, so putting them side by side is not looking good. I believe it's a bug?

```
<FlexLayout>
  <Button>Button</Button>
  <ToggleButtonGroup>
    <ToggleButton aria-label="CSV" tooltipText="CSV">
      <CsvIcon />
    </ToggleButton>
    <ToggleButton aria-label="Loader" tooltipText="Loader">
      <LoaderIcon />
    </ToggleButton>
  </ToggleButtonGroup>
</FlexLayout>
```

Reproduce link: https://stackblitz.com/edit/react-ts-bljppo?file=App.tsx 

Potential fix `+2px` move into bracket?
```
.uitk-density-medium {
  --uitk-size-toggleGroupButton: calc(
    var(--uitk-size-base) - (var(--uitk-size-unit) * 0.5 + 2px)
  );
}
```

### PR DESCRIPTION
**Pull request by joshwooding**
Closes #597

### CODE REVIEW

**Comment by origami-z** [comment]
> release-pr on 

🦋  success packages published successfully:
🦋  @salt-ds/ag-grid-theme@0.0.0-snapshot-20230517152535
🦋  @salt-ds/lab@0.0.0-snapshot-20230517152535
🦋  @salt-ds/theme@0.0.0-snapshot-20230517152535

**Comment by origami-z** [comment]
> 🦋 success packages published successfully: 🦋 @salt-ds/lab@0.0.0-snapshot-20230517152535

Tested above snapshot release (3d old), no breaking change, standalone ToggleButton as well as ToggleButtonGroup works as expected, except keyboard left/right on group doesn't react with below code. It could be due to code is old (given it works on latest Storybook preview), or i'm using it controlled.

```
  const [selectedIndex, setSelectedIndex] = useState<number>(0);

  const handleChange: ToggleButtonGroupChangeEventHandler = (_, index) => {
    setSelectedIndex(index);
  };

...

            <ToggleButtonGroup
              onChange={handleChange}
              selectedIndex={selectedIndex}
            >
              <ToggleButton
                aria-label="Vertical direction"
                tooltipText="Vertical direction"
              >
                <VerticalIcon />
              </ToggleButton>
              <ToggleButton
                aria-label="Horizontal direction"
                tooltipText="Horizontal direction"
              >
                <HorizontalIcon />
              </ToggleButton>
            </ToggleButtonGroup>
```

File: packages/lab/stories/toggle-button-group/toggle-button-group.doc.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+### Text and Icon
+
+Use icons and text in Toggle Buttons if it's important to visually reinforce the theme of the options or if providing additional context may help decision-making (for examples, selecting a document type or a text style).
+
+- Pass `aria-hidden=”true”` to the icon so its label is not announced by the screen reader.
+- To reduce visual noise and mitigate alignment issues, use icons in Toggle Button labels consistently across all options in the group.
+
+<Canvas>
+  <Story id="lab-toggle-button-group--horizontal" />
```
Comment (Fercas123) [inline packages/lab/stories/toggle-button-group/toggle-button-group.doc.stories.mdx:32]: on focus, the icon and text have a different color, is that intended?

**Review by Fercas123 — COMMENTED** [review]
is there specs for TB? the focus ring seems a bit different to what we are doing in list, probably worth matching the approach

File: packages/lab/stories/toggle-button-group/toggle-button-group.doc.stories.mdx (hunk 1 above)
Comment (joshwooding) [reply packages/lab/stories/toggle-button-group/toggle-button-group.doc.stories.mdx:32]: I think it's up to design to push this but the focus styling should include hover styling

**Review by ivan-calderon — APPROVED** [review]

**Pull request merged**